src/main/java/org/opensearch/security/action/apitokens/ApiToken.java

```diff
@@ -0,0 +1,222 @@
+
+public class ApiToken implements ToXContent {
+    public static final String NAME_FIELD = "name";
+    public static final String JTI_FIELD = "jti";
+    public static final String CREATION_TIME_FIELD = "creation_time";
+    public static final String CLUSTER_PERMISSIONS_FIELD = "cluster_permissions";
+    public static final String INDEX_PERMISSIONS_FIELD = "index_permissions";
+    public static final String INDEX_PATTERN_FIELD = "index_pattern";
+    public static final String ALLOWED_ACTIONS_FIELD = "allowed_actions";
+
+    private String name;
+    private final String jti;
+    private final Instant creationTime;
+    private List<String> clusterPermissions;
+    private List<IndexPermission> indexPermissions;
+
+    public ApiToken(String name, String jti, List<String> clusterPermissions, List<IndexPermission> indexPermissions) {
+        this.creationTime = Instant.now();
+        this.name = name;
+        this.jti = jti;
+        this.clusterPermissions = clusterPermissions;
+        this.indexPermissions = indexPermissions;
+
+    }
+
+    public ApiToken(
+        String description,
+        String jti,
+        List<String> clusterPermissions,
+        List<IndexPermission> indexPermissions,
+        Instant creationTime
+    ) {
+        this.name = description;
+        this.jti = jti;
+        this.clusterPermissions = clusterPermissions;
+        this.indexPermissions = indexPermissions;
+        this.creationTime = creationTime;
+
+    }
+
+    public static class IndexPermission implements ToXContent {
+        private final List<String> indexPatterns;
+        private final List<String> allowedActions;
+
+        public IndexPermission(List<String> indexPatterns, List<String> allowedActions) {
+            this.indexPatterns = indexPatterns;
+            this.allowedActions = allowedActions;
+        }
+
+        public List<String> getAllowedActions() {
+            return allowedActions;
+        }
+
+        public List<String> getIndexPatterns() {
+            return indexPatterns;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.array(INDEX_PATTERN_FIELD, indexPatterns.toArray(new String[0]));
+            builder.array(ALLOWED_ACTIONS_FIELD, allowedActions.toArray(new String[0]));
+            builder.endObject();
+            return builder;
+        }
+    }
+
+    public static ApiToken fromXContent(XContentParser parser) throws IOException {
+        String name = null;
+        String jti = null;
+        List<String> clusterPermissions = new ArrayList<>();
+        List<IndexPermission> indexPermissions = new ArrayList<>();
+        Instant creationTime = null;
+
+        XContentParser.Token token;
+        String currentFieldName = null;
+
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
```

Comment:
mind adding a comment above the while loop about what a pre-parsed doc would look like?

Reply:
Good idea added in: [58cf8ca](https://github.com/opensearch-project/security/pull/4921/commits/58cf8cac4696ee0e0c8ea6eb58eea10f1fe7a460)